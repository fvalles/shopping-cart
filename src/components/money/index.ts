export { Money } from './money';
export { TypographyType } from './money';
