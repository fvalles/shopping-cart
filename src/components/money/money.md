Basic money component with default currency="â‚¬":

```jsx
<Money amount={200} typographyType={TypographyType.LIST_ITEM_BIG} />
```

Money component with custom currency property set:

```jsx
<Money
  amount={200}
  currency="$"
  typographyType={TypographyType.LIST_ITEM_SMALL}
/>
```
