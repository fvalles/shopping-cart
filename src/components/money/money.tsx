import React, { FunctionComponent } from 'react';
import { Div } from '../div';
import { Span } from '../typography';

/**
 * Types
 */

interface MoneyProps {
  /** Amount of money to be displayed */
  amount: number;
  /** Currency being used */
  currency?: string;
  /** Component wrapper width */
  width: string;
}

/**
 * Money
 */

export const Money: FunctionComponent<MoneyProps> = ({
  amount,
  currency = 'â‚¬',
  width,
}) => (
  <Div display="flex" width={width} justifyContent="center">
    <Span>{amount}</Span>
    <Span margin="0px 0px 0px 4px">{currency}</Span>
  </Div>
);
