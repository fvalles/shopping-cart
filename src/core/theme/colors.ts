import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  primary: palette.venetianNights,
  productImageBorder: palette.sharpGrey,
  productPrice: palette.black,
  productQuantity: palette.faintFern,
  productTitle: palette.mischka,
  summaryBackground: palette.bleachedSilk,
  summaryHeading: palette.stormGrey,
  summaryText: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
