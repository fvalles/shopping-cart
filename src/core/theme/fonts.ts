export const Fonts = {
  light: 'Avenir Light',
  medium: 'Avenir Medium',
  regular: 'Avenir Regular',
};

export type FontsType = typeof Fonts;
