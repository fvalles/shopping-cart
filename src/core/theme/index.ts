export { Colors } from './colors';
export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
