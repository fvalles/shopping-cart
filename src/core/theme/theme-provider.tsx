import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Colors, Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Colors,
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
