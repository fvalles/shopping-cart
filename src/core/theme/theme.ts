import { DefaultTheme } from 'styled-components';
import { Colors } from './colors';

const theme: DefaultTheme = {
  Colors,
};

export { theme };
