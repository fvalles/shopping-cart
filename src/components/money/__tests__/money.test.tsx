import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Money, TypographyType } from '../money';

describe('money component', () => {
  it('should render amount and currency inside span component', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Money amount={200} />
        </ThemeProvider>,
      )
      .toJSON();

    const amountComponent = result[0];
    expect(amountComponent).toHaveStyleRule('color', '#000000');
    expect(amountComponent.type).toBe('span');
    expect(amountComponent.children[0]).toBe('200');
    const currencyComponent = result[1];
    expect(currencyComponent).toHaveStyleRule('color', '#000000');
    expect(currencyComponent).toHaveStyleRule('margin', '0px 0px 0px 4px');
    expect(currencyComponent.children[0]).toBe('â‚¬');
    expect(result).toMatchSnapshot();
  });

  it('should render amount and currency inside list item big component with custom currency', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Money
            amount={400}
            currency="$"
            typographyType={TypographyType.LIST_ITEM_BIG}
          />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#212240');
    expect(result).toHaveStyleRule('font-weight', 'bold');
    expect(result).toHaveStyleRule('font-size', '20px');
    expect(result.type).toBe('span');
    expect(result.children[0]).toBe('400');
    const currencyComponent = result.children[1];
    expect(currencyComponent).toHaveStyleRule('margin', '0');
    expect(currencyComponent.children[0]).toBe('$');
  });

  it('should render amount and currency inside list item medium component with custom currency', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Money
            amount={400}
            currency="$"
            typographyType={TypographyType.LIST_ITEM_MEDIUM}
          />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#212240');
    expect(result).toHaveStyleRule('font-weight', 'bold');
    expect(result).toHaveStyleRule('font-size', '16px');
    expect(result.type).toBe('span');
    expect(result.children[0]).toBe('400');
    const currencyComponent = result.children[1];
    expect(currencyComponent).toHaveStyleRule('margin', '0px 0px 0px 4px');
    expect(currencyComponent.children[0]).toBe('$');
  });

  it('should render amount and currency inside list item small component with custom currency', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Money
            amount={600}
            currency="$"
            typographyType={TypographyType.LIST_ITEM_SMALL}
          />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#212240');
    expect(result).toHaveStyleRule('font-weight', 'bold');
    expect(result).toHaveStyleRule('font-size', '14px');
    expect(result.type).toBe('span');
    expect(result.children[0]).toBe('600');
    const currencyComponent = result.children[1];
    expect(currencyComponent).toHaveStyleRule('margin', '0px 0px 0px 2px');
    expect(currencyComponent.children[0]).toBe('$');
  });
});
