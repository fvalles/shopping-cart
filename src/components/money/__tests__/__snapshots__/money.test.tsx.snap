// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`money component should render amount and currency inside span component 1`] = `
Array [
  .c0 {
  color: #000000;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 16px;
  line-height: 1;
}

<span
    className="c0"
  >
    200
  </span>,
  .c0 {
  color: #000000;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 16px;
  line-height: 1;
  margin: 0px 0px 0px 4px;
}

<span
    className="c0"
  >
    â‚¬
  </span>,
]
`;
